msctl

```diff
@@ -993,6 +993,54 @@ getServerCommand() {
     printf "$SERVER_LOCATION\n"
     exit 1
   fi
+  # Get the location of the Java virtual machine, use the default value if not provided.
+  DEFAULT_JAVAVM=$(getDefaultsValue "mscs-default-java" "$DEFAULT_JAVA")
+  JAVA=$(getMSCSValue "$1" "mscs-java" "$DEFAULT_JAVAVM")
+  if [ ! -e "$JAVA" ]; then
+    echo "ERROR: java not found!"
+    echo "Try installing this with:"
+    echo "sudo apt-get install default-jre"
+    exit 1
+  fi
+  # Verify that the Java virtual machine supports the server version.
+  JAVA_VERSION=$($JAVA -version 2>&1 | $PERL -ne 'if ($_ =~ /version\s"([\d\.]+)\.\d+\_\d+"/) { print $1 }')
+  case $JAVA_VERSION in
+    1.9)
+      # Java 9 supports all server versions.
+    ;;
+    1.8)
+      # Java 8 supports all server versions.
+    ;;
+    1.7)
+      # Support for Java 7 ended with version 1.12.
+      VERSION_TEST=$(compareMinecraftVersions $SERVER_VERSION 1.12.1)
+      if [ $VERSION_TEST -ge 0 ]; then
```

Comment:
This should be:
```
if [ $VERSION_TEST -gt 0 ]; then
```

I was just wondering about snapshot versions. Do all 1.12 snapshot versions support Java 6 and 7 but only the final release, 1.12.1 drop support?  Maybe it would be best to assume that snapshot versions have the same requirements as the release versions.  In that case, maybe we should do the following comparison:
```
VERSION_TEST=$(compareMinecraftVersions $SERVER_VERSION 1.11.2)
if [ $VERSION_TEST -le 0 ]; then
```

Reply:
I think you're right. MC Wiki says Release 1.12 is the first version to use/require Java 8, so I think it's safe to assume that the snapshots (which are < Release 1.12) are probably Java 6/7 compatible.

Reply:
Actually, according to the [MC wiki](https://minecraft.gamepedia.com/17w13a), 17w13a (the 2nd snapshot) is the first version to require Java 8. Since I don't want to hard code snapshot versions into the script, I think the modified test I posted (<= 1.11.2) would be the most appropriate.